src/pybind/mgr/dashboard_v2/frontend/src/app/core/auth/login/login.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit, ViewContainerRef } from '@angular/core';
+import { Login } from './login.model';
+import { AuthService } from '../../../shared/services/auth.service';
+import { Router } from '@angular/router';
+import { HttpClient } from '@angular/common/http';
+import { ToastsManager } from 'ng2-toastr';
+
+@Component({
+  selector: 'oa-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+
+  model = new Login();
+
+  constructor(private authService: AuthService,
+              private router: Router,
+              private http: HttpClient,
+              public toastr: ToastsManager,
+              private vcr: ViewContainerRef) {
+    this.toastr.setRootViewContainerRef(vcr);
+  }
+
+  ngOnInit() {
+    if (this.authService.isLoggedIn()) {
+      this.router.navigate(['']);
+    }
+  }
+
+  login() {
+    this.http.post(`/api/auth`, this.model).subscribe((data: any) => {
```

Comment:
I believe the best practices say that the HTTP request should be done within the `AuthService` code.
For example add a method to `AuthService` similar to this:

```
import 'rxjs/add/operator/toPromise';

/* ... */

login(credentials: Login): Promise<any> {
    return this.http.post('/api/auth', credentials).toPromise().then(resp => {
        let data = resp.json();
        this.set(data.username);
    })
}
```

Then here you can call the service like this:

```
this.authService.login(this.model).then(resp => {
  this.router.navigate([''])
});
```

Reply:
Done